tools/sense_studio/training.py

```diff
@@ -40,68 +33,88 @@ def train_model():
                         f"--num_layers_to_finetune={num_layers_to_finetune}",
-                        "--use_gpu" if config['use_gpu'] else "",
                         f"--path_out={path_out}" if path_out else "",
                         f"--model_name={model_name}" if model_name else "",
                         f"--model_version={model_version}" if model_version else "",
                         f"--epochs={epochs}",
+                        f"--path_annotations_train={path_annotations_train}" if config['temporal'] else "",
+                        f"--path_annotations_valid={path_annotations_valid}" if config['temporal'] else "",
+                        "--use_gpu" if config['use_gpu'] else "",
+                        f"--temporal_training" if config['temporal'] else "",
                         "--overwrite"]
 
     train_classifier = ' '.join(train_classifier)
     train_classifier = shlex.split(train_classifier)
 
-    global PROCESS
-    PROCESS = subprocess.Popen(train_classifier, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=False)
-
-    def get_training_logs():
-        while True:
-            global PROCESS
-            output = PROCESS.stdout.readline()
-            if output == b'' and PROCESS.poll() is not None:
-                PROCESS.terminate()
-                PROCESS = None
-                break
-            if output:
-                time.sleep(0.1)
-                yield output.decode().strip() + '\n'
+    global train_process
+    train_process = subprocess.Popen(train_classifier, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=False)
 
-    return flask.Response(stream_with_context(stream_template('training.html', project=project, path=path,
-                                                              is_disabled=is_disabled,
-                                                              models=utils.BACKBONE_MODELS,
-                                                              logs=get_training_logs())))
+    return render_template('training.html', project=project, path=path, models=utils.BACKBONE_MODELS,
+                           is_disabled=True, output_folder=output_folder, message="Train Model...")
 
 
 @training_bp.route('/cancel-training', methods=['POST'])
 def cancel_training():
     data = request.form
     project = data['project']
     path = data['path']
-    global PROCESS
-    if PROCESS:
-        PROCESS.terminate()
-        PROCESS = None
-        log = "Training Cancelled."
+    output_folder = data['output_folder']
+    global train_process
+    if train_process:
+        train_process.terminate()
+        train_process = None
+        message = "Training Cancelled."
+    return render_template('training.html', project=project, path=path, models=utils.BACKBONE_MODELS,
+                           is_disabled=False, message=message, output_folder=output_folder)
+
+
+@socketio.on('training_logs', namespace='/train-model')
```

Comment:
Can we give this message another name? I took me some time to understand that this is the "connect" message from the client

Reply:
Technically, it's an event name `training_logs` likewise `connect`. This one method works as connection, logs retriever, and at last disconnect sender.

Reply:
yes, if it bothers more we can think about something else